src/observability/publisher.go

```diff
@@ -20,19 +20,19 @@ func ProcessSystemSummary() {
 	tempSummarizerMap := common.MoveMap(SummarizerMap)
 	SummarizerMapMutex.Unlock()
 
-	var ProcessSystemSummaryWg sync.WaitGroup
+	aggregateSummaryMap(tempSummarizerMap)
 
 	if cfg.GetCfgObservabilityWriteToDB() {
+		var ProcessSystemSummaryWg sync.WaitGroup
 		ProcessSystemSummaryWg.Add(1)
 		go UpsertSummaryCronJob(tempSummarizerMap, &ProcessSystemSummaryWg)
+		ProcessSystemSummaryWg.Wait()
```

Comment:
We can remove the entire ProcessSystemSummaryWg by removing the call to `UpsertSummaryCronJob` as a goroutinr